### PR TITLE
Quantile Regression Chapter

### PR DESCRIPTION
**Pull request by lona-k**

### CODE REVIEW

File: book/chapters/chapter16/quantile_regression.qmd [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+  geom_point(data = data[sample(.N, size = 0.1 * .N)], aes(x = x, y = y), alpha = 0.6) +
+  geom_line(data = true_data, aes(x = x, y = y), color = "red", linewidth = 1) +
+  theme_minimal()
+```
+
+This plot reveals two important properties of our data. Firstly, we can see that the structure of the underlying function exhibits more oscillations for small values of `x` but becomes smoother for larger values. Secondly, the plot shows the heteroskedasticity in our data as the variance of `y` increases as `x` grows.
+
+Because of the heteroskedasticity, mean-based models will struggle to provide robust predictions, especially for larger values of `x`, as they will be heavily influenced by extreme deviations. In contrast, the median (0.5-quantile) provides a more stable estimate, while other quantiles (e.g., 0.05 and 0.95) allow us to estimate uncertainty and extreme outcomes.
+
+Now that we have generated our data set, we transform it into a `r ref("TaskRegr")` and split it into a train and test set for modeling.
```
Comment (be-marc) [inline book/chapters/chapter16/quantile_regression.qmd:65]: Just write regression task. We don't need links to these basic objects.

**Review by be-marc — CHANGES_REQUESTED** [review]
Looks already good. Please use these two guides to revise https://github.com/mlr-org/mlr3book/issues/434 and https://github.com/mlr-org/mlr3book/issues/435

File: book/chapters/chapter16/quantile_regression.qmd [hunk 2]
```diff
@@ -0,0 +1,260 @@
...
+
+But the predicted quantiles appear overly jagged and spiky, which suggests that the model is overfitting to the noise in the training data rather than capturing smooth trends. The median estimate oscillates around the true function but does not consistently align with it. The reason for these limitations lies in how random forests construct quantiles. They are estimated from terminal node distributions, which may lead to discontinuous or highly variable predictions.
+
+### Smooth Additive Model with PipeOpLearnerQuantiles {#sec-quantile-qgam}
+
+To address the limitations that we observed with the random regression forest, we will now consider an alternative method: quantile regression with smooth generalized additive models (GAM) using `r ref("mlr3extralearners::LearnerRegrQGam")` from `r ref_pkg("mlr3extralearners")`. This approach allows for smoother estimates and may improve the robustness of quantile predictions. Unlike tree-based methods, GAMs offer more flexibility by modeling the relationship between the features and the target using smooth functions rather than discrete splits. This makes them well-suited for handling continuous and structured data.
```
[the remark this answers is not in the document]
Comment (lona-k) [reply book/chapters/chapter16/quantile_regression.qmd:151]: Does this apply to every time I reference a learner like this?

**Comment by berndbischl** [comment]
somewhat unfortunate where we currently are in the outline.
i know we want separate chaps with "+".
but this really is "beyond classic and regr"
a) at least put it next to it
b) but as we are adding new subchaps at some point too, I would likely put this into "beyond" with a "+"

File: book/chapters/chapter16/quantile_regression.qmd [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+
+{{< include ../../common/_setup.qmd >}}
+
+`r chapter = "Quantile Regression (+)"`
+`r authors(chapter)`
+
+```{r}
+# Extra packages that must be installed in the docker image
+remotes::install_github("mlr-org/mlr3extralearners")
+remotes::install_cran("qgam")
```
Comment (berndbischl) [inline book/chapters/chapter16/quantile_regression.qmd:16]: this looks weird

Comment (lona-k) [reply book/chapters/chapter16/quantile_regression.qmd:16]: This is necessary right now to properly test the chapter in the PR, but we are only using it as long as this branch is not integrated into the main branch of mlr3book
